### PR TITLE
[yul-phaser] Temporarily disable very slow tests for the classic algorithm

### PR DESCRIPTION
**Pull request by cameel**
This disables the 5 tests that take 99% of `soltest` execution time with `--run_test=Phaser`.

### CODE REVIEW

File: test/yulPhaser/GeneticAlgorithms.cpp [hunk 1]
```diff
@@ -212,6 +212,8 @@ BOOST_FIXTURE_TEST_CASE(runNextRound_should_generate_individuals_in_the_crossove
 BOOST_AUTO_TEST_SUITE_END()
 BOOST_AUTO_TEST_SUITE(ClassicGeneticAlgorithmTest)
 
+// FIXME: These tests run *very* slowly (tens of seconds). Investigate, fix and re-enable them.
+/*
 BOOST_FIXTURE_TEST_CASE(runNextRound_should_select_individuals_with_probability_proportional_to_fitness, ClassicGeneticAlgorithmFixture)
```
Comment (axic) [inline test/yulPhaser/GeneticAlgorithms.cpp]: Does boost have something like a `DISABLED_` prefix (similar to gtest) which places these disabled from running but still compiling?

Comment (cameel) [reply test/yulPhaser/GeneticAlgorithms.cpp]: Good point. Looks like there is. There's the `disable()` decorator. I changed the code to use it.

One downside of it is that disabled tests get re-enabled when you use filters on the command line, which I often do (e.g. `--run_test=Phaser`): [Re: [Boost-users] Boost Unit Test: How to run only the enabled tests of a test suite?](https://lists.boost.org/boost-users/2017/12/88215.php)
Fortunately these tests are not actually broken, just slow so in this case it's fine but it would not work if we wanted to completely disable them.

**Review by ekpyron — APPROVED** [review]
How much time buys this in the CI run :-)?

**Comment by cameel** [comment]
On my local debug build it's 6:29 before vs 5:36 after. So 57 seconds less.
The longest soltest run when [CircleCI was building the current release](https://app.circleci.com/pipelines/github/ethereum/solidity/6197/workflows/e0d9b26d-d706-4e16-a243-8e7762c9e96c) took 41:55 so it could cut it down to something like 35:19. Improvement should be noticeable but it's still going to be pretty slow.

**Review by axic — APPROVED** [review]

**Comment by cameel** [comment]
Soooo... the funny thing is that looking at recent successful builds, it's hard to say for sure that this PR actually improves CI times :)

Here's a table of a few recent successful runs on Circle CI (basically all the all-green workflows from the first 3 pages):

|   CI run                                                                                                                     |   Branch                                                        |   Total duration   |   t_ubu_asan_constantinople   |   t_ubu_release_soltest   |   t_ubu_soltest   |   t_osx_soltest   |   t_ubu_clang_soltest   |   t_ubu_soltest_enforce_yul   |
|------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------|--------------------|-------------------------------|---------------------------|-------------------|-------------------|-------------------------|-------------------------------|
| **[#6211](https://app.circleci.com/pipelines/github/ethereum/solidity/6211/workflows/71170294-2592-4a18-9b08-7d315e5236a0)** | **yul-phaser-temporarily-disable-slow-classic-algorithm-tests** | **1h 36m 14s**     | **1h 22m 10s**                | **44m 7s**                | **33m 16s**       | **4m 31s**        | **2m 15s**              | **3m 17s**                    |
|   [#6179](https://app.circleci.com/pipelines/github/ethereum/solidity/6179/workflows/563a27e0-281e-4e67-baf4-26592425b484)   |   refactor-sideEffects                                          |   1h 50m 45s       |   1h 31m 10s                  |   46m 35s                 |   43m 30s         |   4m 38s          |   2m 32s                |   3m 38s                      |
|   [#6195](https://app.circleci.com/pipelines/github/ethereum/solidity/6195/workflows/d419c34c-479b-418e-8f5f-82fd7d1267da)   |   propagateArguments                                            |   1h 58m 12s       |   1h 42m 29s                  |   34m 11s                 |   36m 29s         |   4m 14s          |   3m  3s                |   2m 19s                      |
|   [#6187](https://app.circleci.com/pipelines/github/ethereum/solidity/6187/workflows/46ce0090-3c70-4d84-889d-7ba46db210a9)   |   interface-events                                              |   1h 58m 37s       |   1h 44m 29s                  |   33m 2s                  |   39m 52s         |   4m 41s          |   2m 11s                |   3m 2s                       |
|   [#6212](https://app.circleci.com/pipelines/github/ethereum/solidity/6212/workflows/9b450ecd-1518-43e9-aa32-0d524c1bb86b)   |   fix-9308                                                      |   1h 59m 50s       |   1h 44m 58s                  |   31m 42s                 |   29m 32s         |   4m 52s          |   3m 37s                |   2m 17s                      |
|   [#6196](https://app.circleci.com/pipelines/github/ethereum/solidity/6196/workflows/10901309-70ae-4dcf-baec-19fe0090a500)   |   compileTimeSha                                                |   2h 3m 14s        |   1h 48m 17s                  |   34m 13s                 |   28m 51s         |   4m 23s          |   2m 37s                |   3m 17s                      |
|   [#6197](https://app.circleci.com/pipelines/github/ethereum/solidity/6197/workflows/e0d9b26d-d706-4e16-a243-8e7762c9e96c)   |   release                                                       |   2h 5m 13s        |   1h 50m 4s                   |   34m 3s                  |   41m 55s         |   4m 30s          |   2m 32s                |   2m 44s                      |
|   [#6194](https://app.circleci.com/pipelines/github/ethereum/solidity/6194/workflows/930a5847-7c87-4112-82a3-95fa1978458b)   |   versionTo0612                                                 |   2h 5m 17s        |   1h 44m 24s                  |   32m 3s                  |   32m 2s          |   4m 39s          |   2m 21s                |   2m 28s                      |
|   [#6209](https://app.circleci.com/pipelines/github/ethereum/solidity/6209/workflows/bffb1528-5759-437e-a4e0-9e740618c5b5)   |   issue-3412                                                    |   2h 11m 58s       |   1h 57m 13s                  |   30m 6s                  |   29m 55s         |   4m 41s          |   2m 59s                |   3m 32s                      |
|   [#6170](https://app.circleci.com/pipelines/github/ethereum/solidity/6170/workflows/a8fa7eb4-5347-4e4c-bd08-9d95437bfa98)   |   noVisibilityForConstructor                                    |   2h 16m 57s       |   1h 57m 48s                  |   35m 9s                  |   28m 59s         |   4m 35s          |   2m 22s                |   3m 9s                       |
|   [#6169](https://app.circleci.com/pipelines/github/ethereum/solidity/6169/workflows/ddd218d3-4a73-4606-9060-8f3d45c51417)   |   fixOverridingWording                                          |   2h 19m 42s       |   1h 58m 33s                  |   30m 39s                 |   56m 2s          |   4m 28s          |   2m 22s                |   2m 52s                      |
|   [#6186](https://app.circleci.com/pipelines/github/ethereum/solidity/6186/workflows/43972a24-391b-4f4a-ac56-4d603b72d036)   |   issue-8911                                                    |   2h 30m 1s        |   1h 47m 44s                  |   36m 29s                 |   31m 44s         |   4m 35s          |   4m 30s                |   4m 10s                      |
|   [#6198](https://app.circleci.com/pipelines/github/ethereum/solidity/6198/workflows/e9019e99-2d53-4a83-8d2f-89c1826b4ead)   |   issue-8911                                                    |   2h 31m 28s       |   2h 17m 32s                  |   31m 10s                 |   42m 25s         |   4m 46s          |   2m 10s                |   3m 17s                      |
|   [#6185](https://app.circleci.com/pipelines/github/ethereum/solidity/6185/workflows/3401e191-e608-4df5-9483-fefe96bdde97)   |   prepare0611                                                   |   2h 31m 37s       |   2h 13m 22s                  |   28m 48s                 |   37m 52s         |   4m 19s          |   3m 33s                |   2m 13s                      |
|   [#6191](https://app.circleci.com/pipelines/github/ethereum/solidity/6191/workflows/c0171522-9766-4219-b748-65025fff7951)   |   develop                                                       |   2h 37m 59s       |   2h 18m 44s                  |   30m 9s                  |   55m 36s         |   4m 25s          |   2m 24s                |   2m 1s                       |
|   [#6190](https://app.circleci.com/pipelines/github/ethereum/solidity/6190/workflows/dd9c7701-8dfb-4708-84f3-677697e1666d)   |   develop                                                       |   2h 38m 5s        |   2h 22m 58s                  |   31m 30s                 |   35m 38s         |   4m 21s          |   2m 23s                |   2m 41s                      |

Basically, soltest run times are all over the place in successful builds. It's hard to attribute the difference to the change in phaser's tests given the variance. The only job where my fix seems to have clearly sped things up is `t_ubu_asan_constantinople`. But we only have a sample of 1 run so it could just have been lucky...

The fix does have an observable and deterministically reproducible effect when running tests locally in debug so it's worth merging anyway but gains in CI aren't clear. I am going to retrigger the tests to have another sample but I think it won't be very conclusive. Something else is affecting the tests way more.

**Comment by chriseth** [comment]
I'll merge it - if it improves running time locally, it's already a win.
